lib/concurrent/synchronization.rb

```diff
@@ -6,23 +6,26 @@
 require 'concurrent/synchronization/rbx_object'
 
 module Concurrent
+  # {include:file:doc/synchronization.md}
   module Synchronization
-    class Object < case
-                           when Concurrent.on_jruby?
-                             JavaObject
+    Implementation = case
+                     when Concurrent.on_jruby?
+                       JavaObject
+                     when Concurrent.on_cruby? && (RUBY_VERSION.split('.').map(&:to_i) <=> [1, 9, 3]) <= 0
```

Comment:
I'd like to extract `(RUBY_VERSION.split('.').map(&:to_i) <=> [1, 9, 3]) <= 0` into a predicate method on `Engine`. You've done a great job making our code easier to reason about with methods like `Concurrent.on_jruby?`.